### PR TITLE
RELATED: FET-534 Tiger wiremock tests

### PR DESCRIPTION
**Pull request by plavemar**
- sdk-backend-tiger base for wiremock integrated tests
- scripts to run tests, create recordings for tests, authentication
- catalog tests for tiger backend

JIRA: FET-534

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6280/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8074/ : FAILURE in 43s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6281/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8075/ : FAILURE in 39s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6282/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8076/ : FAILURE in 41s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6288/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8086/ : FAILURE in 16m 49s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6289/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8087/ : FAILURE in 42s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6290/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8088/ : FAILURE in 16m 33s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6291/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8089/ : FAILURE in 23m 58s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6295/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8093/ : FAILURE in 49s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6296/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8094/ : FAILURE in 17m 14s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6310/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8107/ : FAILURE in 53s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6311/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8108/ : FAILURE in 18m 57s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6312/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8109/ : SUCCESS in 18m 31s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6316/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8113/ : SUCCESS in 24m 09s

File: libs/sdk-backend-tiger/package.json [hunk 1]
```diff
@@ -60,9 +69,11 @@
         "ts-invariant": "^0.7.3",
         "tslib": "^2.0.0",
         "url": "^0.11.0",
-        "uuid": "^8.3.2"
+        "uuid": "^8.3.2",
+        "dotenv": "^10.0.0"
```
Comment (no23reason) [inline libs/sdk-backend-tiger/package.json:73]: Please keep this in alphabetical order.

File: libs/sdk-backend-tiger/scripts/refresh-md.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+SCRIPTDIR=`dirname $0`
+ROOTDIR="${SCRIPTDIR}/.."
+EXPORTER="${ROOTDIR}/node_modules/.bin/gdc-catalog-export"
+OUTPUT="${ROOTDIR}/src/fixtures/full.ts"
+# todo create a reference workspace for tiger wiremock tests with GoodSales fixtures
```
Comment (no23reason) [inline libs/sdk-backend-tiger/scripts/refresh-md.sh:7]: This TODO is no longer needed., right?

File: libs/sdk-backend-tiger/src/fixtures/referenceObjects.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// (C) 2022 GoodData Corporation
+import { ObjRef } from "@gooddata/sdk-model";
+
+export const stageStatuses = [
+    { title: "Lost", uri: "/api/elements/referenceworkspace/obj/1100/elements?id=460496" },
```
Comment (no23reason) [inline libs/sdk-backend-tiger/src/fixtures/referenceObjects.ts:5]: The uri values are strange, on tiger they are often the same as title, i.e. they almost never look like uris.

File: libs/sdk-backend-tiger/tests/integrated/insights.test.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// (C) 2022 GoodData Corporation
+// (C) 2020 GoodData Corporation
```
Comment (no23reason) [inline libs/sdk-backend-tiger/tests/integrated/insights.test.ts:2]: Duplicate header.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2068

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1823/ : SUCCESS in 1m 46s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2104/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/905/ : SUCCESS in 16m 44s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8117/ : SUCCESS in 21m 05s

**Pull request merged**